Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -195,66 +195,50 @@
-%   1 control volume and 3 or 7 PLDs, respectively). So:
+% With Hadamard-encoded ASL, the fully encoded Hadamard matrix is separated in several (e.g., 4 or 8) blocks corresponding to Hadamard-4 or Hadamard-8.
+% This means that before each acquisition we alternate 4 or 8 blocks of control/label as specified by the Hadamard matrix.
+% By block-wise subtraction and addition of the 4 or 8 acquired volumes according to the Hadamard encoding matrix, 
+% we can decode 1 control volume and 3 or 7 PLDs, respectively.
 %
-% block = control or single PLD acquisition encoded in Hadamard
 % TimeEncodedMatrixSize = nBlocks
 
 nBlocks = xQ.TimeEncodedMatrixSize;
 
-%   INNER REPETITIONS
-%   There may be readout-related reasons for having multiple volumes.
-%   E.g., with a multi-TE readout, each control or label volume (in this case each Hadamard-encoded volume) is acquired multiple 
-%   times for the number of echoes. Because this is repeated at the readout level, we define these here as "inner repetitions".
-%   E.g., for a 8 echoes Hadamard acquisition, a single control or PLD-encoded acquisition is repeated 8 times, so:
+% INNER REPETITIONS
+% Normally, for a 4-block Hadamard, we acquire 4 volumes. But there may be readout-related reasons for having more volumes than that.
+% E.g., with a multi-TE readout, each Hadamard-encoded volume is acquired multiple times equal to the number of echoes. Because this is repeated 
+% at the readout level, we define these here as "inner repetitions". E.g., for a 8-echoes Hadamard acquisition, a single control or PLD-encoded acquisition is repeated 8 times.
 %
 % innerRepetition: single-volume repetitions within a Hadamard block
 
 [~, ~, indexUniquePLD] = unique(xQ.Initial_PLD); % Find volume indices of unique PLDs
-indexFirstPLD = find(indexUniquePLD==1); % Find the volumes of the first PLD
-indexSecondPLD = find(indexUniquePLD==2); % Find the volumes of the second PLD
+indexSecondPLD = find(indexUniquePLD==2); % Find the volumes of the second PLD - all PLDs before the second PLD are equivalent
 if isempty(indexSecondPLD)
 	nInnerRepetitions = length(xQ.Initial_PLD); % There's no second PLD and nInnerRepetitions is the length of the PLD vector
 else
-	nInnerRepetitions = indexSecondPLD - 1; % The number of first-PLDs, i.e. the count of all PLDs before the second one, is the number of inner repetitions
+	nInnerRepetitions = indexSecondPLD(1) - 1; % The number of first-PLDs, i.e. the count of all PLDs before the second PLD is the number of inner repetitions
 end
 
 if nInnerRepetitions ~= xQ.NumberEchoTimes % Check for TE number differ from the number of inner repeats as this is a special option needing a special reordering
-	warning('Number of inner repetitions and echo-times does not match, such a sequence is not yet implemented');
+	warning('Number of inner repetitions and echo-times does not match for a Hadamard sequence. Quantification of such a sequence is not yet implemented');
```

Comment:
`nVolumesPerInnerRepetition` is confusing here, because a single inner repetition is only within a single block right? E.g., 8 echoes with 4 Hadamard blocks should have 8 nVolumesPerInnerRepetition, unless AFTER your reshaping, when the Hadamard blocks are inside the InnerRepetition?

So now I think I get it:

`nVolumesPerInnerRepetition` should be `nVolumesPerOuterRepetition`

and

`nAcquisitionRepetitions` should be `nOuterRepetitions`

right?

Reply:
You created `nAcquisitionRepetitions` and `nOuterRepetitions`, while these were equivalent. I've picked up the first and used only that one. But I'm fine changing to the second.

`nVolumesPerInnerRepetition` I'm fine to have it as you've changed it.

So you can resolve this.